engine/collection/message_hub/message_hub_test.go

```diff
@@ -0,0 +1,376 @@
+		err := s.hub.Process(channel, originID, syncedBlockMsg)
+		require.NoError(s.T(), err)
+
+		blockProposalMsg := &messages.ClusterBlockProposal{
+			Header:  block.Header,
+			Payload: block.Payload,
+		}
+		s.compliance.On("Process", channel, s.myID, blockProposalMsg).Return(nil).Once()
+		err = s.hub.Process(channel, originID, blockProposalMsg)
+		require.NoError(s.T(), err)
+	})
+	s.Run("to-vote-aggregator", func() {
+		expectedVote := unittest.VoteFixture(unittest.WithVoteSignerID(originID))
+		msg := &messages.ClusterBlockVote{
+			View:    expectedVote.View,
+			BlockID: expectedVote.BlockID,
+			SigData: expectedVote.SigData,
+		}
+		s.voteAggregator.On("AddVote", expectedVote)
+		err := s.hub.Process(channel, originID, msg)
+		require.NoError(s.T(), err)
+	})
+	s.Run("to-timeout-aggregator", func() {
+		expectedTimeout := helper.TimeoutObjectFixture(helper.WithTimeoutObjectSignerID(originID))
+		msg := &messages.ClusterTimeoutObject{
+			View:       expectedTimeout.View,
+			NewestQC:   expectedTimeout.NewestQC,
+			LastViewTC: expectedTimeout.LastViewTC,
+			SigData:    expectedTimeout.SigData,
+		}
+		s.timeoutAggregator.On("AddTimeout", expectedTimeout)
+		err := s.hub.Process(channel, originID, msg)
+		require.NoError(s.T(), err)
+	})
+	s.Run("unsupported-msg-type", func() {
+		err := s.hub.Process(channel, originID, struct{}{})
+		require.NoError(s.T(), err)
+	})
+}
+
+// TestBroadcastProposalWithDelay tests broadcasting proposals with different inputs
+func (s *MessageHubSuite) TestBroadcastProposalWithDelay() {
+	// add execution node to cluster to make sure we exclude them from broadcast
+	s.cluster = append(s.cluster, unittest.IdentityFixture(unittest.WithRole(flow.RoleExecution)))
+
+	// generate a parent with height and chain ID set
+	parent := unittest.ClusterBlockFixture()
+	parent.Header.ChainID = "test"
+	parent.Header.Height = 10
+
+	// create a block with the parent and store the payload with correct ID
+	block := unittest.ClusterBlockWithParent(&parent)
+	block.Header.ProposerID = s.myID
+
+	s.headers.On("ByBlockID", block.Header.ParentID).Return(parent.Header, nil)
+	s.headers.On("ByBlockID", mock.Anything).Return(nil, storerr.ErrNotFound)
+	s.payloads.On("ByBlockID", block.Header.ID()).Return(block.Payload, nil)
+	s.payloads.On("ByBlockID", mock.Anything).Return(nil, storerr.ErrNotFound)
+
+	s.Run("should fail with wrong proposer", func() {
+		header := *block.Header
+		header.ProposerID = unittest.IdentifierFixture()
+		err := s.hub.processQueuedProposal(&header)
+		require.Error(s.T(), err, "should fail with wrong proposer")
+		header.ProposerID = s.myID
+	})
+
+	// should fail with changed (missing) parent
+	s.Run("should fail with changed/missing parent", func() {
+		header := *block.Header
+		header.ParentID[0]++
+		err := s.hub.processQueuedProposal(&header)
+		require.Error(s.T(), err, "should fail with missing parent")
+		header.ParentID[0]--
+	})
```

Comment:
We should remove this after incorporating https://github.com/onflow/flow-go/pull/3372#pullrequestreview-1145645406